src/fetchers/Chain/ChainUserFetcher.ts

```diff
@@ -106,4 +111,31 @@ export class ChainUserFetcher extends ChainFetcher implements UserFetcher {
   ) {
     return this._provider.getBalance(userAddress, blockTag);
   }
+
+  async fetchManagerApproval(
+    userAddress: Address,
+    managerAddress: Address,
+    blockTag: BlockTag = "latest"
+  ) {
+    return this._morpho!.isManagedBy(userAddress, managerAddress, { blockTag });
+  }
+  async fetchStethData(userAddress: Address, blockTag: BlockTag = "latest") {
+    const stEth = StEth__factory.connect(addresses.steth, this._provider);
+    const [balance, stethPerWsteth, permit2Approval, bulkerApproval] =
+      await Promise.all([
+        stEth.balanceOf(userAddress, {
+          blockTag,
+        }),
+        stEth.getPooledEthByShares(WadRayMath.WAD, { blockTag }),
```

Comment:
are you sure about this function? I thought that `getPooledEthByShares` was used to get the stEth/Eth ratio whereas here we're looking for the stEth/wstEth

Reply:
[Yes, I'm sure](https://github.com/lidofinance/lido-dao/blob/master/contracts/0.6.12/WstETH.sol#L108)

In a nutshell, 1 wsteth = 1eth